ariadne/cardano/src/Ariadne/Wallet/Cardano/WalletLayer/Types.hs

```diff
@@ -94,6 +96,11 @@ data PassiveWalletLayer m = PassiveWalletLayer
     , pwlRollbackBlocks
           :: NewestFirst NE Blund
           -> m ()
+    , pwlCleanAcidDB
```

Comment:
It's currently in the `core API` section, but it's not core API. I think it should be a new section.
And actually now I think it is better to make it part of `passiveWalletLayerComponent`.
`buildComponent "Wallet DB cleaner" (async cleanupAcidState) uninterruptibleCancel`.

Reply:
Though @Pastafarianist may disagree, so I'd wait for his comments. I think the fact that wallet layer uses acid-state is its implementation detail and the necessity to run some helper worker is caused solely by this detail, so I like the idea of hiding it inside `WalletLayer` (and fortunately it's very easy to do, because `ComponentM` allows you to do equivalent of `withAsync`).

Reply:
We have some tasks about hiding acid DB behind wallet layer (e. g. AD-306 and AD-493), but they are about hiding stuff that is already visible, i. e. that wasn't hidden when it was added.
`pwlCleanAcidDB` is a new stuff, so it's better to hide it when it's added, not postpone it. It shouldn't be hard anyway. But let's see what @Pastafarianist thinks first.

Reply:
@gromakovsky I agree. Cleanup is an implementation detail and certainly not a part of core API.